### PR TITLE
Make source-depth a parameter and opt-in

### PR DESCRIPTION
**Pull request by sergiusens**
LP: #1617783

Signed-off-by: Sergio Schvezov <sergio.schvezov@ubuntu.com>

### CODE REVIEW

**Comment by kalikiana** [comment]
What's the rationale for changing the default behavior? It seems to me like I'd want to add source-depths to all my snaps now because it saves time and disk space.

File: snapcraft/internal/sources.py [hunk 1]
```diff
@@ -119,8 +125,9 @@ def download(self):
 class Script(FileBase):
 
     def __init__(self, source, source_dir, source_tag=None,
-                 source_branch=None):
-        super().__init__(source, source_dir, source_tag, source_branch)
+                 source_branch=None, source_depth=None):
+        super().__init__(source, source_dir, source_tag,
+                         source_branch, source_depth)
```
Comment (josepht) [inline snapcraft/internal/sources.py]: Can we call it 'git-source-depth'? Also items 19-21 in Effective Python might give you some ideas for simplifying the base class `__init__()` by popping the accepted kwargs and throwing an exception if there are any leftover options

Comment (sergiusens) [inline snapcraft/internal/sources.py]: All this sounds good, but the bug mentions `source-depth` and this same modus operandi we are doing applies for `source-tag` when using local sources or tarballs

I will leave the popping for when we move the logic to the pluginhandler

**Comment by sergiusens** [comment]
El 20/09/16 a las 11:26, Christian Dywan escribió:
>
> What's the rationale for changing the default behavior? It seems to me 
> like I'd want to add source-depths to all my snaps now because it 
> saves time and disk space.
>
Refer to the bug please for discussing the rationale.

**Comment by sergiusens** [comment]
@kalikiana in case you missed it, refer to the bug please for discussion of the rationale

File: snapcraft/internal/sources.py [hunk 2]
```diff
@@ -33,6 +33,11 @@
     control system or compression algorithm. The source-type key can tell
     snapcraft exactly how to treat that content.
 
+  - source-depth: <integer>
+
+    By default clones or branches with full history, specifying a depth
+    will truncate the history to the specified number of commits.
```
Comment (elopio) [inline snapcraft/internal/sources.py]: What about mentioning that usually only one branch is needed?

**Comment by elopio** [comment]
lgtm

**Comment by kyrofa** [comment]
I'm not a fan of this change, but :+1: to the code.

**Pull request merged**